common/interpreter/formatSwitch.go

```diff
@@ -22,25 +23,48 @@ func NewFormatSwitch(templates ...FormatTemplate) *FormatSwitch {
 	}
 }
 
-// ParseJSON will select one of the templates, populate and return it.
-// If error response can be interpreted concisely and to the point, then great.
-// Otherwise, fallback and use whole response to build an error.
-// This strategy always exposes, never hides what the server sent us.
-func (s FormatSwitch) ParseJSON(data []byte) ErrorDescriptor { // nolint:ireturn
+// ParseJSON selects a template, populates it, and returns the result. If the error response is concise and clear,
+// it’s used directly; otherwise, the entire response is used to build an error.
+// This strategy ensures complete visibility into the server’s response.
+//
+// The JSON data may either contain a single object, interpreted as an ErrorDescriptor,
+// or an array of objects, with each mapped to matching ErrorDescriptor.
+func (s FormatSwitch) ParseJSON(data []byte) ErrorDescriptor {
+	// First: assume HTTP response data is an array of objects.
+	listPayload := make([]map[string]any, 0)
```

Comment:
this not payload, it is `response`


`payload` is a term used when we want to send something as part of a body in a request.